utils/tests/test_ipappliance.py

```diff
@@ -9,7 +9,7 @@ def test_ipappliance_from_address():
     address = '1.2.3.4'
     ip_a = IPAppliance(address)
     assert ip_a.address == address
-    assert ip_a.url == 'https://%s/' % address
```

Comment:
I disagree. IPAppliance.url should instead make sure the URL is predictable, and always has a trailing slash.